lib/krane/deploy_task.rb

```diff
@@ -279,13 +279,17 @@ def validate_configuration(prune:)
 
     def validate_resources(resources)
       validate_globals(resources)
+      dry_run_success = validate_dry_run(resources)
       Krane::Concurrency.split_across_threads(resources) do |r|
-        r.validate_definition(kubectl, selector: @selector)
+        # No need to pass in kubectl (and do per-resource dry run apply) if batch dry run succeeded
+        if dry_run_success
+          r.validate_definition(kubectl: nil, selector: @selector)
```

Comment:
I'd rather add a param than have passing kubectl: nil imply that dry run isn't needed